sdks/java/harness/src/main/java/org/apache/beam/fn/harness/PrecombineGroupingTable.java

```diff
@@ -223,27 +323,26 @@ public void put(
       throws Exception {
     // Ignore timestamp for grouping purposes.
     // The Pre-combine output will inherit the timestamp of one of its inputs.
-    WindowedValue<Object> groupingKey =
-        WindowedValue.of(
-            keyCoder.structuralValue(value.getValue().getKey()),
-            IGNORED,
-            value.getWindows(),
-            value.getPane());
-
-    GroupingTableEntry entry =
-        lruMap.compute(
-            groupingKey,
-            (key, tableEntry) -> {
-              if (tableEntry == null) {
-                tableEntry =
-                    new GroupingTableEntry(
-                        groupingKey, value.getValue().getKey(), value.getValue().getValue());
-              } else {
-                tableEntry.add(value.getValue().getValue());
-              }
-              return tableEntry;
-            });
-    weight += entry.getWeight();
+    GroupingTableKey groupingKey =
+        new GroupingTableKey(
+            value.getValue().getKey(), value.getWindows(), value.getPane(), keyCoder, keySizer);
+
+    lruMap.compute(
+        groupingKey,
+        (key, tableEntry) -> {
+          if (tableEntry == null) {
+            weight += groupingKey.getWeight();
```

Comment:
remove this?

Reply:
this adds the weight of the key, and not the value

Reply:
isn't entry.getWeight() = key.getWeight() + accumulator.getWeight()?

Reply:
There are two cases. 
* key == structural key, then:
  * GroupingTableKey weight = key weight + windows weight + pane info weight
  * GroupingTableEntry weight = reference weight + accumulator weight
* key != structural key, then:
  * GroupingTableKey weight = structural key weight + windows weight + pane info weight
  * GroupingTableEntry weight = key weight + accumulator weight